src/v2/ion/i18nTransformer.js

```diff
@@ -150,6 +150,50 @@ const I18N_PARAMS_MAPPING = {
   },
 };
 
+/**
+ * For i18n keys that require string interpolation using values from "params".
+ * {baseKey} : {params}
+ */
+ const I18N_OVERRIDE_WITH_PARAMS_MAP = {
+  'registration.error.invalidLoginEmail': {
+    Email: 'Email'
+  },
+  'registration.error.doesNotMatchPattern': {
+    Email: 'Email'
+  },
+};
+
+/**
+ * For messages that need to be interpolated with param values.
+ *
+ * Enumerate each possible param interpolation and hardcode that into properties file.
+ * This is to ensure proper translation.
+ *
+ * Example:
+ * input =
+ *  "i18n": {
+      "key": "registration.error.doesNotMatchPattern",
+      "params": [
+        "Email"
+      ]
+    }
+ * output = registration.error.invalidLoginEmail.Email
```

Comment:
nit: can we correct this to `registration.error.doesNotMatchPattern.Email` to match the input example ?